js/views/file_input_view.js

```diff
@@ -177,30 +190,40 @@
             return files && files.length && files.length > 0;
         },
 
-        getFiles: function() {
-            var promises = [];
-            var files = this.file ? [this.file] : this.$input.prop('files');
-            for (var i = 0; i < files.length; i++) {
-                promises.push(this.getFile(files[i]));
-            }
-            this.clearForm();
-            return Promise.all(promises);
-        },
+    /* eslint-enable */
+    /* jshint ignore:start */
+    getFiles() {
+      const files = this.file ? [this.file] : Array.from(this.$input.prop('files'));
+      const promises = Promise.all(
```

Comment:
The value `promises` here doesn't quite make sense anymore, now that is is just a single promise, post `.all()`

Reply:
For clarity, I reversed the order to the original and named it `promise` for clarity: 329036e59c138c1e950ec7c654eebd7d87076de5